tests/unit/db/database_tests.py

```diff
@@ -299,19 +299,28 @@ def test_retrieve_raw_view_results(self):
         self.assertIsInstance(raw_rslt, dict)
         self.assertEqual(len(raw_rslt.get('rows')), 100)
 
-    def test_all_docs(self):
+    def test_all_docs_post(self):
         """
-        Test the all_docs functionality
+        Test the all_docs POST request functionality
+        """
+        keys_list = list(map(lambda x: x['id'], self.populate_db_with_documents()))
```

Comment:
- Now using a subset of keys for the documents in the test database.
- Also added tests for POST with `keys` alone and POST with `keys` and other arguments.

Done in 82e9646